test/versioned/openai/openai.tap.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper')
+const createOpenAIMockServer = require('../../lib/openai-mock-server')
+// TODO: remove config once we fully release OpenAI instrumentation
+const config = {
+  feature_flag: {
+    openai_instrumentation: true
+  }
+}
+
+tap.test('OpenAI instrumentation', (t) => {
+  t.autoend()
+
+  t.before(async () => {
+    const { host, port, server } = await createOpenAIMockServer()
+    t.context.server = server
+    t.context.agent = helper.instrumentMockedAgent(config)
+    const OpenAI = require('openai')
+    t.context.client = new OpenAI({
+      apiKey: 'fake-versioned-test-key',
+      baseURL: `http://${host}:${port}`
+    })
+  })
+
+  t.afterEach(() => {
+    t.context.agent.customEventAggregator.clear()
+  })
+
+  t.teardown(() => {
+    t.context?.server?.close()
+    t.context.agent && helper.unloadAgent(t.context.agent)
+  })
```

Comment:
This is interesting. It never occurred to me that the parent test's `testObject` would have `.context` defined. I assumed it was only added to subtest objects. This approach is definitely equivalent to:

```js
tap.test('something), async t => {
  let server
  let agent

  t.before(() => {
    server = {}
    agent = {}
  })
})
``` 

But what it provides over that pattern is that it prevents current and future tests from defining variables that shadow any such local scoped variables. 👍